app/models/vm_or_template/operations.rb

```diff
@@ -118,6 +118,16 @@ def validate_vm_control
       msg ||= _("Provider doesn't support vm_destroy") unless ext_management_system.respond_to?(:vm_destroy)
       unsupported_reason_add(:terminate, msg) if msg
     end
+
+    supports :suspend do
+      msg = unsupported_reason(:control) unless supports_control?
+      msg ||= _("The VM is not powered on") unless vm_powered_on?
+      unsupported_reason_add(:suspend, msg) if msg
+    end
+  end
+
+  def vm_powered_on?
+    current_state == "on"
```

Comment:
after move it to vm/operations/power.rb
rspec ./spec/models/manageiq/providers/redhat/infra_manager/vm_spec.rb[1:1:3:1] # ManageIQ::Providers::Redhat::InfraManager::Vm#is_available? with :suspend when powered on
rspec ./spec/models/manageiq/providers/openstack/cloud_manager/vm_spec.rb[1:2:3:1] # ManageIQ::Providers::Openstack::CloudManager::Vm#is_available? with :suspend when powered on
rspec ./spec/requests/api/instances_spec.rb:269 # Instances API Instance suspend action suspends multiple instances
rspec ./spec/requests/api/instances_spec.rb:243 # Instances API Instance suspend action cannot suspend a powered off instance
rspec ./spec/requests/api/instances_spec.rb:252 # Instances API Instance suspend action cannot suspend a suspended instance
rspec ./spec/requests/api/instances_spec.rb:261 # Instances API Instance suspend action suspends an instance
rspec ./spec/models/manageiq/providers/vmware/cloud_manager/vm_spec.rb[1:1:3:1] # ManageIQ::Providers::Vmware::CloudManager::Vm#is_available? with :suspend when powered on
rspec ./spec/requests/api/vms_spec.rb:283 # Vms API Vm suspend action suspends a vm
rspec ./spec/requests/api/vms_spec.rb:291 # Vms API Vm suspend action suspends multiple vms
rspec ./spec/requests/api/vms_spec.rb:265 # Vms API Vm suspend action suspends a powered off vm
rspec ./spec/requests/api/vms_spec.rb:274 # Vms API Vm suspend action suspends a suspended vm
rspec ./spec/models/manageiq/providers/vmware/infra_manager/vm_spec.rb[1:1:3:1] # ManageIQ::Providers::Vmware::InfraManager::Vm#is_available? with :suspend when powered on

the problem that most part of providers inherit from ::Vm class that inherit from VmOrTemplate